### PR TITLE
Updating recommended email settings for monolog

### PR DESCRIPTION
**Pull request by jorgelbg**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | >=2.3 |
| Fixed tickets | #6467 |

### CODE REVIEW

File: cookbook/logging/monolog_email.rst [hunk 1]
```diff
@@ -92,26 +97,29 @@ it is broken down.
...
                     // 'excluded_404s' => array(
                     //     '^/',
                     // ),
-                    'handler'      => 'buffered',
+                    'handler'      => 'deduplicated',
                 ),
-                'buffered' => array(
-                    'type'    => 'buffer',
-                    'handler' => 'swift',
+                'duplicated' => array(
```
Comment (javiereguiluz) [inline cookbook/logging/monolog_email.rst]: Should be `deduplicated` here?

File: cookbook/logging/monolog_email.rst [hunk 2]
```diff
@@ -49,33 +51,36 @@ it is broken down.
...
             <monolog:config>
-                <!--
-                To also log 400 level errors (but not 404's):
+                <!-- 
+                500 errors are logged at the critical level,
+                to also log 400 level errors (but not 404's):
                 action-level="error"
                 And add this child inside this monolog:handler
-                <monolog:excluded-404>^/</monolog:excluded-404>
+                <monolog:excluded-404>^/</monolog:excluded-404>
```
Comment (xabbuh) [inline cookbook/logging/monolog_email.rst]: Can you revert this change (the extra space)?

**Comment by Txumari** [comment]
I'm new here, but I think should be change the buffered explanation with the new handler deduplicated (The ``deduplicated`` handler simply keeps...)

**Comment by jorgelbg** [comment]
Fixed the typos, and amended the commit. Thanks @javiereguiluz @xabbuh

**Comment by xabbuh** [comment]
👍

**Comment by javiereguiluz** [comment]
👍

(by the way, I really dislike the "deduplication" word ... but that's something we cannot change)

**Comment by WouterJ** [comment]
I agree with @Txumari here, after this code block we talk about the buffered handler a couple times. We should change it to deduplicated handler (and also change the description).

**Comment by jorgelbg** [comment]
Should we mantain some information about the `buffer` handler? its still a valid handler according to https://github.com/Seldaek/monolog/blob/master/doc/02-handlers-formatters-processors.md#wrappers--special-handlers but honestly the `deduplication` handler does also the buffering + deduplication of messages. The final example use the `group` and `buffer` ?

**Comment by jorgelbg** [comment]
I've updated the explanation and fixed the mentions of the `buffered` handler, also changed the last example in the doc, that talked about combining the handlers to store the messages on the server and also send the logs by email.

File: cookbook/logging/monolog_email.rst [hunk 3]
```diff
@@ -121,20 +129,23 @@ it is only triggered when the action level, in this case ``critical`` is reached
...
-handler then each message will be emailed separately. This is then passed
-to the ``swift`` handler. This is the handler that actually deals with
-emailing you the error. The settings for this are straightforward, the
-to and from addresses and the subject.
+The ``deduplicated`` handler simply keeps all the messages for a request and
+then passes them onto the nested handler in one go, but only if the records are
+unique over a given period of time (60 seconds by default). If the records are
+duplicates they are simply discarded. Adding this handler reduces the amount of
+notifications to a manageable level, specially in critical failure scenarios.
+The messages are then passed to the ``swift`` handler. This is the handler that
```
Comment (xabbuh) [inline cookbook/logging/monolog_email.rst]: I would add a blank line before this sentence to have it rendered as its own paragraph to improve readability.

Comment (jorgelbg) [inline cookbook/logging/monolog_email.rst]: Done!

**Comment by WouterJ** [comment]
:+1: I like it

Status: Reviewed

**Comment by xabbuh** [comment]
Thank you @jorgelbg.
